modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -36,109 +35,117 @@ class ContentBuilder implements ContentBuilderInterface {
    */
-  protected function getTopics(BlockContent $block_content) {
-    // Get topic type tags.
-    $topic_types_list = $block_content->get('field_topic_type')->getValue();
-    $topic_types = array_map(function ($topic_type) {
-      return $topic_type['target_id'];
-    }, $topic_types_list);
-
-    // Get group tags.
-    $group_tag_list = $block_content->get('field_group')->getValue();
-    $group_tags = array_map(function ($group_tag) {
-      return $group_tag['target_id'];
-    }, $group_tag_list);
-
-    // Get social tags.
-    $social_tag_list = $block_content->get('field_content_tags')->getValue();
-    $social_tags = array_map(function ($social_tag) {
-      return $social_tag['target_id'];
-    }, $social_tag_list);
-
-    // Use database select because we need joins
-    // which are not possible with entityQuery.
-    /** @var \Drupal\Core\Database\Query\SelectInterface $query */
-    $query = $this->connection->select('node_field_data', 'n')
-      ->fields('n', ['nid'])
-      ->condition('n.type', 'topic');
-
-    // Add topic type tags.
-    if (!empty($topic_types)) {
-      $query->innerJoin('node__field_topic_type', 'tt', 'tt.entity_id = n.nid');
-      $query->condition('tt.field_topic_type_target_id', $topic_types, 'IN');
+  protected function getEntities(BlockContentInterface $block_content) {
+    $plugin_id = $block_content->field_plugin_id->value;
+    $definition = $this->contentBlockManager->getDefinition($plugin_id);
+
+    if ($block_content->field_plugin_field->isEmpty()) {
+      $field_names = $definition['fields'];
+    }
+    else {
+      $field_names = [$block_content->field_plugin_field->value];
     }
 
-    // Add group tags.
-    if (!empty($group_tags)) {
-      $query->innerJoin('group_content_field_data', 'gd', 'gd.entity_id = n.nid');
-      $query->condition('gd.gid', $group_tags, 'IN');
+    $fields = [];
+
+    foreach ($field_names as $field_name) {
+      $fields[$field_name] = array_map(function ($item) {
+        return $item['target_id'];
+      }, $block_content->get($field_name)->getValue());
     }
 
-    if (!empty($social_tags)) {
-      $query->innerJoin('node__social_tagging', 'st', 'st.entity_id = n.nid');
-      $query->condition('st.social_tagging_target_id', $social_tags, 'IN');
+    /** @var \Drupal\social_content_block\ContentBlockPluginInterface $plugin */
+    $plugin = $this->contentBlockManager->createInstance($plugin_id);
+
+    /** @var \Drupal\Core\Entity\EntityTypeInterface $entity_type */
+    $entity_type = $this->entityTypeManager->getDefinition($definition['entityTypeId']);
+
+    /** @var \Drupal\Core\Database\Query\SelectInterface $query */
+    $query = $this->connection->select($entity_type->getBaseTable(), 'base_table')
+      ->fields('base_table', [$entity_type->getKey('id')]);
+
+    if ($definition['bundle']) {
+      $query->condition(
+        'base_table.' . $entity_type->getKey('bundle'),
```

Comment:
Is the table name of condition sanitized? 🤔

Reply:
What do you mean?